hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -363,6 +372,37 @@ public RecordsRunningHashLeaf runningHashLeaf() {
 		return metadata.getUniqueTreasuryOwnershipAssociations();
 	}
 
+	private void updateLinks() {
+		var accounts = accounts();
+		var tokenRels = tokenAssociations();
+
+		for (var accountId : accounts.keySet()) {
+			var merkleAccount = accounts.getForModify(accountId);
+			int numAssociations = 0;
+			int numZeroBalances = 0;
+			EntityNumPair prevListRootKey = MISSING_NUM_PAIR;
+			for (var tokenId : merkleAccount.tokens().asTokenIds()) {
+				var newListRootKey = EntityNumPair.fromLongs(accountId.longValue(), tokenId.getTokenNum());
+				var association = tokenRels.getForModify(newListRootKey);
+				association.setNextKey(prevListRootKey);
+
+				if (prevListRootKey != MISSING_NUM_PAIR) {
+					var prevAssociation = tokenRels.getForModify(prevListRootKey);
+					prevAssociation.setPrevKey(newListRootKey);
```

Comment:
I might be missing something, but should this be 
`newListRootKey.setPrevKey(prevAssocation)` ?

Trying to trace this logic out and I see that it'll look like this at the end:
```
  prevListRootKey
     |
    v
[token N-1] <-[...] <-[token 2] <- [token 1] <- [token 0] 
```